src/main/java/de/cronn/validation_files_diff/helper/PsiTestNameUtils.java

```diff
@@ -0,0 +1,28 @@
+package de.cronn.validation_files_diff.helper;
+
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiMethod;
+
+public class PsiTestNameUtils {
+
+	private PsiTestNameUtils() {
+	}
+
+	public static String getTestName(PsiClass psiClass, PsiMethod psiMethod) {
+		return join(enclosingClassesUpstream(psiClass), psiMethod.getName());
+	}
+
+	private static String enclosingClassesUpstream(PsiClass psiClass) {
+		String classHierarchy = psiClass.getName();
+		PsiClass enclosingClass = psiClass.getContainingClass();
+		while (enclosingClass != null) {
+			classHierarchy = join(enclosingClass.getName(), classHierarchy);
+			enclosingClass = enclosingClass.getContainingClass();
+		}
+		return classHierarchy;
```

Comment:
Maybe avoid the term "class hierarchy" that is usually used for the inheritance and not the nesting hierarchy? Some suggestions: `result`, `nestingHierarchy`, `nestingPath`, ...

Reply:
done, fdb9eeb9d36624d5e72091f4efd9761ab52f86ac